### ISSUE TITLE
Broken type definition in focus-trap-react 11

### ISSUE THREAD

**Issue by Mathias-S** [focus-trap/focus-trap-react#1396]
**Description**
I am using ESM-style imports and exports in my project.

The change from `export = FocusTrap` to `export declare class FocusTrap` in [index.d.ts](https://github.com/focus-trap/focus-trap-react/commit/4a37daeade3c4d43da9e045441ff974d8acf1072#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8L4) means that it is now only possible to use _named_ imports, i.e. `import { FocusTrap } from "react-focus-trap"` (according to the types).

However, this is incompatible with `module.exports = FocusTrap;` in  [src/focus-trap-react.js](https://github.com/focus-trap/focus-trap-react/blob/4a37daeade3c4d43da9e045441ff974d8acf1072/src/focus-trap-react.js#L428), which only lets you do _default_ imports, i.e.  `import FocusTrap from "react-focus-trap"`.

So the type change needs to be reverted so only default imports works, or a named export should be provided instead:
```js
exports.FocusTrap = FocusTrap;
```

But that would obviously also be a breaking change. It is technically possible to hack it and provide both a named and default export with CJS:
```js
module.exports = FocusTrap;
module.exports.FocusTrap = FocusTrap;
```

**Versions**
- focus-trap-react version used: 11.0.0

**Sandbox**
Minimal reproduction in CodeSandbox should display the type error in the editor: https://codesandbox.io/p/sandbox/5fz5r4

**Local repro**
```bash
npm create vite@latest focus-trap-test -- --template react-ts
cd focus-trap-test
npm install focus-trap-react
# then edit `src/App.tsx` to add FocusTrap. If you have the TS language server running, your editor should highlight the error
```

**Comment by stefcameron** [comment]
@Mathias-S I like that idea! At least it's progress toward the goal as I (ever so) slowly refactor this library. I'd love to have it be 100% TypeScript some day instead of have "tangential" types like it does today.

**Comment by stefcameron** [comment]
@Mathias-S I have opened #1406 to do as you suggested (and closed #1397): Provide both exports and mark the default export as deprecated. I tested this in your local `vite` repro and it works as intended. Please have a look and LMK if I missed something -- particularly with the types.

**Comment by stefcameron** [comment]
@all-contributors add @Mathias-S for bug, review
